webextension/background.js

```diff
@@ -361,11 +361,13 @@ let window_height = {};
 // due to a race condition with messages.
 function send_to_overlay(id, msg) {
     if (!overlay_ready[id]) {
-	deferred_message[id] = msg;
+	    deferred_message[id] = msg;
+    }
+    else {
+        if (window_height[id]) // add latest known height
```

Comment:
adding else to avoid error => if the overlay is not ready then defer and no need to send message.

Reply:
Drop the else branch, the "blind" send is intentional: getting the addon to work correctly as soon as a tab is loaded (there are/can be corner cases where this could be noticeable to the user) is more important than needlessly spamming error logs, especially now that the errors are silenced.

Reply:
I've removed the else branch but in my understanding, a tab loaded is a tab with status `completed`.
So you could then test the state of the tab and then send or not the message.
You could do something like:
```
async function send_to_overlay(id, msg) {
    if (!overlay_ready[id]) {
        deferred_message[id] = msg;
    }
    if (window_height[id]) // add latest known height
        msg.mainwindow_height = window_height[id];

    const tab = await browser.tabs.get(id);
    if tab.status === "completed" {
        browser.tabs.sendMessage(id, msg).catch(e => {});
    }
}
```

Reply:
The tab.status=completed could only mean DOM completed (and does that include iframe DOMs as well?...), not whether all necessary scripts have run, especially if the scripts have asynchronous parts. Also the note in https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/Tab seems to say that it is not available to background scripts anyway.

Reply:
> not whether all necessary scripts have run, especially if the scripts have asynchronous parts. 
You're right, not sure about this so it's safer to continue as you do.